tools/doc-generator/doc-generator.go

```diff
@@ -21,32 +18,34 @@ import (
 )
 
 // constant parts of the file
-const (
-	genFileComment = `<!--
-	This is an auto-generated file.
-	PLEASE DO NOT EDIT THIS FILE.
-	See "Developing new metrics" below how to generate this file
--->`
-	title      = "# KubeVirt metrics\n"
-	background = "This document aims to help users that are not familiar with all metrics exposed by different KubeVirt components.\n" +
-		"All metrics documented here are auto-generated by the utility tool `tools/doc-generator` and reflects exactly what is being exposed.\n\n"
-
-	KVSpecificMetrics = "## KubeVirt Metrics List\n" +
-		"### kubevirt_info\n" +
-		"Version information.\n\n"
-
-	opening = genFileComment + "\n\n" +
-		title +
-		background +
-		KVSpecificMetrics
-
-	// footer
-	footerHeading = "## Developing new metrics\n"
-	footerContent = "After developing new metrics or changing old ones, please run `make generate` to regenerate this document.\n\n" +
-		"If you feel that the new metric doesn't follow these rules, please change `doc-generator` with your needs.\n"
-
-	footer = footerHeading + footerContent
-)
+const tpl = `# Kubevirt metrics
+
+### kubevirt_info
+Version information.
+
+{{- range . }}
+
+{{ $deprecatedVersion := "" -}}
+{{- with index .ExtraFields "DeprecatedVersion" -}}
+    {{- $deprecatedVersion = printf " in %s" . -}}
+{{- end -}}
+
+{{- $stabilityLevel := "" -}}
+{{- if and (.ExtraFields.StabilityLevel) (ne .ExtraFields.StabilityLevel "STABLE") -}}
+	{{- $stabilityLevel = printf "[%s%s] " .ExtraFields.StabilityLevel $deprecatedVersion -}}
+{{- end -}}
+
+### {{ .Name }}
+{{ print $stabilityLevel }}{{ .Help }} Type: {{ .Type -}}.
+
+{{- end }}
+
+## Developing new metrics
+
+All metrics documented here are auto-generated and reflect exactly what is being
+exposed. After developing new metrics or changing old ones please regenerate
+this document.
+`
 
 func main() {
 	handler := domainstats.Handler(1)
```

Comment:
this is only used for domainstats metrics, but this will be part of the virt-handler metric listing after the refactor, this code should also be removed IMO, and this PR only merged after that refactor

Reply:
@machadovilaca can you mentioned the refactor pr's that need to be merged before this one?